### PR TITLE
Remove Flaky annotation from TestHiveStorageFormats#testInsertIntoTable, rely on QueryExecutors retry mechanism

### PR DESCRIPTION
**Pull request by homar**
Replace query with onTrino.executeQuery to make sure that retry mechanism from QueryExecutors is used

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 1]
```diff
@@ -292,7 +292,6 @@ public void verifyDataProviderCompleteness()
     }
 
     @Test(dataProvider = "storageFormatsWithConfiguration", groups = {STORAGE_FORMATS, HMS_ONLY})
-    @Flaky(issue = ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE, match = ERROR_COMMITTING_WRITE_TO_HIVE_MATCH)
```
Comment (losipiuk) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:295]: have two separate commits. One which moves to `onTrino()`. The other which removes `@Flaky` annotation.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 2]
```diff
@@ -25,7 +25,6 @@
 import io.trino.tempto.query.QueryExecutionException;
 import io.trino.tempto.query.QueryExecutor.QueryParam;
 import io.trino.tempto.query.QueryResult;
-import io.trino.testng.services.Flaky;
```
Comment (losipiuk) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:28]: should be part of second commit

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 3]
```diff
@@ -63,12 +62,9 @@
 import static io.trino.tempto.assertions.QueryAssert.assertThat;
 import static io.trino.tempto.query.QueryExecutor.defaultQueryExecutor;
 import static io.trino.tempto.query.QueryExecutor.param;
-import static io.trino.tempto.query.QueryExecutor.query;
 import static io.trino.tests.product.TestGroups.HMS_ONLY;
 import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
 import static io.trino.tests.product.TestGroups.STORAGE_FORMATS_DETAILED;
-import static io.trino.tests.product.hive.HiveProductTest.ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE;
```
Comment (losipiuk) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:70]: same here

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 4]
```diff
@@ -63,12 +62,9 @@
...
-import static io.trino.tests.product.hive.HiveProductTest.ERROR_COMMITTING_WRITE_TO_HIVE_MATCH;
```
Comment (losipiuk) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:71]: same here

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java (hunk 1 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:295]: make title of second commit slightly shorter, and put some context in the message (why the `@Flaky`  is no longer needed)

**Review by losipiuk — COMMENTED** [review]
LGTM. Minor comments.

**Comment by homar** [comment]
@losipiuk  should be better now

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
Thx

**Pull request merged**